seatunnel-translation/seatunnel-translation-flink/seatunnel-translation-flink-20/src/main/java/org/apache/seatunnel/translation/flink/metric/FlinkAccumulatorCounter.java

```
@@ -0,0 +1,172 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.seatunnel.translation.flink.metric;
+
+import org.apache.seatunnel.api.common.metrics.Counter;
+import org.apache.seatunnel.api.common.metrics.MetricNames;
+import org.apache.seatunnel.api.common.metrics.Unit;
+
+import org.apache.flink.api.common.accumulators.LongCounter;
+import org.apache.flink.api.common.functions.RuntimeContext;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class FlinkAccumulatorCounter implements Counter {
+    private final String name;
+    private final org.apache.flink.metrics.Counter flinkCounter;
+    private final LongCounter accumulator;
+    private final RuntimeContext runtimeContext;
+    private volatile long localCount = 0L;
+
+    public FlinkAccumulatorCounter(
+            String name,
+            org.apache.flink.metrics.Counter flinkCounter,
+            RuntimeContext runtimeContext) {
+        this.name = name;
+        this.flinkCounter = flinkCounter;
+        this.runtimeContext = runtimeContext;
+        this.accumulator = new LongCounter();
+
+        try {
+            String accumulatorName = getStandardAccumulatorName(name);
+            runtimeContext.addAccumulator(accumulatorName, accumulator);
+        } catch (Exception e) {
+            log.warn("Failed to register accumulator: {}", name);
+        }
+    }
+
+    @Override
+    public void inc() {
+        inc(1L);
+    }
+
+    @Override
+    public void inc(long n) {
+        try {
+            if (flinkCounter != null) {
+                flinkCounter.inc(n);
+            }
+
+            accumulator.add(n);
+
+            localCount += n;
+
+        } catch (Exception e) {
+            log.warn("Error incrementing counter {}", name);
+            localCount += n;
+        }
```

why catch Exception? We should direct throw it.

---

> why catch Exception? We should direct throw it.

Removed the exception-swallowing try/catch logic from inc/dec/set/getCount/sync, and propagate exceptions directly when encountered.